lib/experimental/Lists/DataList/ItemWithCopyButton.tsx

```diff
@@ -0,0 +1,78 @@
+import { useEffect, useState } from "react"
+
+import { Icon } from "@/components/Utilities/Icon"
+import { ItemProps } from "@/experimental/Lists/DataList"
+import { ItemContainer } from "@/experimental/Lists/DataList/ItemContainer"
+import { CheckCircle, LayersFront } from "@/icons"
+import { cn } from "@/lib/utils"
+
+const COPIED_SHOWN_MS = 750
+
+type CopyButtonProps = ItemProps
+
+export const ItemWithCopyButton = ({
```

Comment:
Just playing devils advocate, could there be a case where we'd want a different item type to also be able to copy its content? 🤔

Make the "copy" action an item itself could be limiting or confusing in the future.

Also, will always the shown text be the one we want to copy to the clipboard?

Reply:
I initially went the generic way, growing the complexity until I found out we don't have such requirement yet. It prefer to expand the functionality when needed, rather than delivering a flexible solution and support it